pagerduty/resource_pagerduty_event_orchestration_path_service.go

```diff
@@ -240,29 +277,69 @@ func resourcePagerDutyEventOrchestrationPathServiceUpdate(d *schema.ResourceData
 	}
 
 	payload := buildServicePathStruct(d)
+	serviceID := payload.Parent.ID
 	var servicePath *pagerduty.EventOrchestrationPath
 
-	log.Printf("[INFO] Creating PagerDuty Event Orchestration Service Path: %s", payload.Parent.ID)
+	log.Printf("[INFO] Creating PagerDuty Event Orchestration Service Path: %s", serviceID)
 
 	retryErr := resource.Retry(30*time.Second, func() *resource.RetryError {
-		if path, _, err := client.EventOrchestrationPaths.Update(payload.Parent.ID, "service", payload); err != nil {
+		if path, _, err := client.EventOrchestrationPaths.Update(serviceID, "service", payload); err != nil {
 			return resource.RetryableError(err)
 		} else if path != nil {
 			d.SetId(path.Parent.ID)
 			servicePath = path
 		}
 		return nil
 	})
-
 	if retryErr != nil {
 		return retryErr
 	}
 
 	setEventOrchestrationPathServiceProps(d, servicePath)
 
+	if needToUpdateServiceActiveStatus(d) {
+		enableEOForService := d.Get("enable_event_orchestration_for_service").(bool)
+		log.Printf("[INFO] Updating PagerDuty Event Orchestration Path Service Active Status for service: %s", serviceID)
+
+		retryErr = resource.Retry(30*time.Second, func() *resource.RetryError {
+			resp, _, err := client.EventOrchestrationPaths.UpdateServiceActiveStatus(serviceID, enableEOForService)
+			if err != nil && isErrCode(err, http.StatusGone) {
+				return nil
+			}
+			if err != nil {
+				time.Sleep(2 * time.Second)
+				return resource.RetryableError(err)
+			}
+			if resp.Active != enableEOForService {
+				time.Sleep(2 * time.Second)
+				return resource.RetryableError(fmt.Errorf("uncosistent result received when trying to update event orchestration active status for service %q", serviceID))
+			}
+			return nil
+		})
+		if retryErr != nil {
+			return retryErr
+		}
+		d.Set("enable_event_orchestration_for_service", enableEOForService)
+	}
+
 	return nil
 }
 
+func needToUpdateServiceActiveStatus(d *schema.ResourceData) bool {
+	var needToUpdate bool
+	if d.HasChange("enable_event_orchestration_for_service") {
+		o, n := d.GetChange("enable_event_orchestration_for_service")
+		old := o.(bool)
+		new := n.(bool)
+		_, ok := d.GetOkExists("enable_event_orchestration_for_service")
+		if ok || old != new && new == false {
+			needToUpdate = true
+		}
```

Comment:
Question (just to make sure I understand what this code does) - does the condition `ok || old != new && new == false` mean that `needToUpdate` will be set to `true` in the following cases:
- if a user has `enable_event_orchestration_for_service` explicitly set to either `true` or `false`
- if a user had `enable_event_orchestration_for_service = true` and then deleted it from their TF config?

In other words if the config looks like this:
```hcl
resource "pagerduty_event_orchestration_service" "serviceA" {
    service = "POOPBUG"
    enable_event_orchestration_for_service = true
    ...
}
```
**OR**:
```hcl
resource "pagerduty_event_orchestration_service" "serviceA" {
    service = "POOPBUG"
    enable_event_orchestration_for_service = false
    ...
}
```

**OR**:
```hcl
# old config:
resource "pagerduty_event_orchestration_service" "serviceA" {
    service = "POOPBUG"
    enable_event_orchestration_for_service = true
    ...
}

# new config:
resource "pagerduty_event_orchestration_service" "serviceA" {
    service = "POOPBUG"
    #  enable_event_orchestration_for_service was deleted
    ...
}
```

Reply:
Sounds good, thanks for confirming!